src/App.js

```diff
@@ -77,10 +81,10 @@ function App() {
       }
     })
 
-    setOnboard(onboard)
+    setOnboard(newOnboard)
 
     setNotify(initNotify())
-  }, [])
+  }, [onboard, setAddress, setOnboard, setNetwork, setNotify])
```

Comment:
Hey @paulrberg! It looks like this might not be necessary because

* the existing dependencies set, `[]`, means `useEffect` will only run the first time;
* `const onboard` inside the `useEffect` body is local, so the state variable's value isn't relevant (renaming the local variable might still be a good idea!)
* `setThing` functions are promised (see [note](https://reactjs.org/docs/hooks-reference.html#usestate)) to have a stable identity across renders, so they are safe to omit from the dependencies list.

Cheers!

Reply:
Oh, I see. Thanks for clarifying!